### DISCUSSION TITLE
simultaneous, always available, two-way audio

### DISCUSSION THREAD

**Discussion by weow147** [pschatzmann/arduino-audio-tools#1754] [Q&A] [answered]
It doesn't look like it's possible to me, but i'm hoping to be proven wrong....

Picture an iphone with both an external microphone and external speaker plugged into it via lightning (or usbc).
In any given app, you would notice that you can input audio thru the external mic, and output audio thru your external speaker.
Now, make that wireless.
That's what i need.

I thought the hands-free profile (HFP) might work, but it's not implemented, and it seems you can only get the two-way audio happening during a phone call.

Thoughts?

**Comment by pschatzmann** [comment, marked as answer]
You've lost me with your description.

The ESP32 is supporting HFP and there is even a [demo app in the IDF](https://github.com/espressif/esp-idf/blob/master/examples/bluetooth/bluedroid/classic_bt/hfp_hf/README.md)!

**Comment by weow147** [comment]
i found that demo, but i dismissed it for 3 reasons.  1) it's categorized under bluedroid which reminds me of android, my case is apple 2) the first ref is someting.py and i was reminded of python, which i want to avoidand 3) it seems to live in the esspressif idf which i'm unfamiliar with.  none of that adds up to the Arduino Audio Tools Library in the Arduino IDE.

**Comment by pschatzmann** [comment]
- Bluedroid is the Bluetooth stack that Espressif has licenced to provide Bluetooth functionality.
- Yes you would first try it out in IDF to figure out if it is doing what you expect and to learn output the necessary configuration.
- IDF is part of the ESP32 Arduino stack, so you can easily port it to Arduino: just move the functionality in main() to setup().
